test/servinge2e/servicemesh_test.go

```diff
@@ -0,0 +1,155 @@
+		t.Skipf("test namespace \"%s\" not a mesh member, use \"make install-mesh\" for ServiceMesh setup", serviceMeshTestNamespaceName)
+	}
+
+	tests := []struct {
+		name string
+		annotations map[string]string
+		expectIstioSidecar bool
+	}{{
+		// Requests go via gateway -> activator -> pod , by default
+		// Verifies the activator can connect to the pod
+		name: "sidecar-via-activator",
+		annotations: map[string]string{
+			"sidecar.istio.io/inject":          "true",
+		},
+		expectIstioSidecar: true,
+	},{
+		// Requests go via gateway -> pod ( activator should be skipped if burst capacity is disabled and there is at least 1 replica)
+		// Verifies the gateway can connect to the pod directly
+		name: "sidecar-without-activator",
+		annotations: map[string]string{
+			"sidecar.istio.io/inject":											"true",
+			"autoscaling.knative.dev/targetBurstCapacity": 	"0",
+			"autoscaling.knative.dev/minScale": 						"1",
+		},
+		expectIstioSidecar: true,
+	},{
+		// Verifies the "sidecar.istio.io/inject" annotation is really what decides the istio-proxy presence
+		name: "no-sidecar",
+		annotations: map[string]string{
+			"sidecar.istio.io/inject":					"false",
+		},
+		expectIstioSidecar: false,
+	}}
+
+	for _, scenario := range tests {
+		scenario := scenario
+		t.Run(scenario.name, func(t *testing.T) {
+			// Create a ksvc with the specified annotations
+			service, err := caCtx.Clients.Serving.ServingV1().Services(serviceMeshTestNamespaceName).Create(
+				ksvcWithAnnotations(scenario.name, serviceMeshTestNamespaceName, serviceMeshTestImage, scenario.annotations))
+
+			if err != nil {
+				t.Fatalf("error creating ksvc: %v", err)
+			}
+
+			// Let the ksvc be deleted after test
+			caCtx.AddToCleanup(func() error {
+				t.Logf("Cleaning up Knative Service '%s/%s'", service.Namespace, service.Name)
+				return caCtx.Clients.Serving.ServingV1().Services(serviceMeshTestNamespaceName).Delete(service.Name, &meta.DeleteOptions{})
+			})
+
+			// Wait until the Ksvc is ready.
+			service, err = test.WaitForServiceState(caCtx, service.Name, service.Namespace, test.IsServiceReady)
+			if err != nil {
+				t.Errorf("error waiting for ksvc readiness: %v", err)
+				return
+			}
+
+			// Verify the service is actually accessible from the outside
+			if _, err := pkgTest.WaitForEndpointState(
+				&pkgTest.KubeClient{Kube: caCtx.Clients.Kube},
+				t.Logf,
+				service.Status.URL.URL(),
+				pkgTest.EventuallyMatchesBody(helloworldText),
+				"WaitForRouteToServeText",
+				true); err != nil {
+				t.Errorf("the Route at domain %s didn't serve the expected text \"%s\": %v", service.Status.URL.URL(), helloworldText, err)
+			}
+
+			// Verify the expected istio-proxy is really there
+			podList, err := caCtx.Clients.Kube.CoreV1().Pods(serviceMeshTestNamespaceName).List(meta.ListOptions{LabelSelector: "serving.knative.dev/service=" + service.Name})
+			if err != nil {
+				t.Fatalf("error listing pods: %v", err)
```

Comment:
If `Fatalf` is used, it executes other scenario. So `Errorf` is better.

Reply:
I have changed it to `Errorf` ( and added `return` where it doesn't make sense to continue the current scenario)  

The test job use -failfast though, so it won't run the other scenario anyway if one fails...